java-checks/src/main/java/org/sonar/java/checks/helpers/RegexTreeHelper.java

```diff
@@ -51,37 +49,35 @@ private RegexTreeHelper() {
 
   public static List<RegexCheck.RegexIssueLocation> getGraphemeInList(List<? extends RegexSyntaxElement> trees) {
     List<RegexCheck.RegexIssueLocation> result = new ArrayList<>();
-    RegexSyntaxElement startGrapheme = null;
-    RegexSyntaxElement endGrapheme = null;
+    List<RegexSyntaxElement> codeUnits = new ArrayList<>();
     for (RegexSyntaxElement child : trees) {
       if (child instanceof CharacterTree) {
         CharacterTree currentCharacter = (CharacterTree) child;
         if (!currentCharacter.isEscapeSequence()) {
           if (!isMark(currentCharacter)) {
-            addCurrentGrapheme(result, startGrapheme, endGrapheme);
-            startGrapheme = child;
-            endGrapheme = null;
-          } else if (startGrapheme != null) {
-            endGrapheme = child;
+            addCurrentGrapheme(result, codeUnits);
+            codeUnits.clear();
+            codeUnits.add(currentCharacter);
+          } else if (!codeUnits.isEmpty()) {
+            codeUnits.add(currentCharacter);
           }
           continue;
         }
       }
-      addCurrentGrapheme(result, startGrapheme, endGrapheme);
-      startGrapheme = null;
-      endGrapheme = null;
+      addCurrentGrapheme(result, codeUnits);
+      codeUnits.clear();
     }
-    addCurrentGrapheme(result, startGrapheme, endGrapheme);
+    addCurrentGrapheme(result, codeUnits);
     return result;
   }
 
   private static boolean isMark(CharacterTree currentChar) {
     return MARK_PATTERN.matcher(currentChar.characterAsString()).matches();
   }
 
-  private static void addCurrentGrapheme(List<RegexCheck.RegexIssueLocation> result, @Nullable RegexSyntaxElement start, @Nullable RegexSyntaxElement end) {
-    if (start != null && end != null) {
-      result.add(new RegexCheck.RegexIssueLocation(start, end, ""));
+  private static void addCurrentGrapheme(List<RegexCheck.RegexIssueLocation> result, List<RegexSyntaxElement> codePoints) {
+    if (codePoints.size() > 1) {
+      result.add(new RegexCheck.RegexIssueLocation(codePoints, ""));
```

Comment:
I always fear when giving a reference to a list to another object constructor (what if we modify again this object here, would it have a consequence in the `RegexIssueLocation` ?)
Looking at the code, I can see that it's not the case, but I feel it might be safer to pass a copy anyway, just in case.